File: python/src/ttio/instrument_config.py
"""``InstrumentConfig`` — human-readable acquisition instrument description."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class InstrumentConfig:
    """Instrument configuration block persisted as HDF5 string attributes.

    All fields default to empty strings because the format spec
    explicitly allows every sub-field to be missing (§3 of
    ``docs/format-spec.md``).

    Parameters
    ----------
    manufacturer : str, default ""
        Instrument manufacturer (e.g. ``"Thermo Fisher Scientific"``).
    model : str, default ""
        Instrument model (e.g. ``"Orbitrap Eclipse"``).
    serial_number : str, default ""
        Serial number.
    source_type : str, default ""
        Ionization source (e.g. ``"ESI"``, ``"MALDI"``).
    analyzer_type : str, default ""
        Mass analyzer (e.g. ``"Orbitrap"``, ``"TOF"``).
    detector_type : str, default ""
        Detector (e.g. ``"inductive"``, ``"electron multiplier"``).

    Notes
    -----
    API status: Stable.

    Cross-language equivalents
    --------------------------
    Objective-C: ``TTIOInstrumentConfig`` · Java:
    ``com.dtwthalion.ttio.InstrumentConfig``.
    """

    manufacturer: str = ""
    model: str = ""
    serial_number: str = ""
    source_type: str = ""
    analyzer_type: str = ""
    detector_type: str = ""

File: java/src/main/java/com/dtwthalion/ttio/InstrumentConfig.java
/*
 * TTI-O Java Implementation
 * Copyright (C) 2026 DTW-Thalion
 * SPDX-License-Identifier: LGPL-3.0-or-later
 */
package com.dtwthalion.ttio;

/**
 * Immutable instrument configuration value class.
 *
 * <p>All fields may be empty strings per format-spec §3 (every
 * sub-field is optional).</p>
 *
 * <p><b>API status:</b> Stable.</p>
 *
 * <p><b>Cross-language equivalents:</b> Objective-C
 * {@code TTIOInstrumentConfig}, Python
 * {@code ttio.instrument_config.InstrumentConfig}.</p>
 *
 * @param manufacturer Instrument manufacturer.
 * @param model        Instrument model.
 * @param serialNumber Serial number.
 * @param sourceType   Ionization source type.
 * @param analyzerType Mass analyzer type.
 * @param detectorType Detector type.
 * @since 0.6
 */
public record InstrumentConfig(
    String manufacturer, String model, String serialNumber,
    String sourceType, String analyzerType, String detectorType) {}
